Riot/Modules/Settings/SettingsViewController.m

```diff
@@ -340,6 +340,15 @@ - (void)finalizeInit
     self.screenTracker = [[AnalyticsScreenTracker alloc] initWithScreen:AnalyticsScreenSettings];
 }
 
+- (void)dealloc {
+    // Remove observers
+    if (pushInfoUpdateObserver)
+    {
+        [[NSNotificationCenter defaultCenter] removeObserver:pushInfoUpdateObserver name:kMXKAccountAPNSActivityDidChangeNotification object:nil];
+        pushInfoUpdateObserver = nil;
+    }
+}
```

Comment:
`SettingsViewController.m` already has the `- (void)reset` method which removes observers.
Why we need to do this twice here?

Reply:
I was searching observer removal based on notification's name.
The instantiated observers are correctly removed (not using notification names) by `reset` method.
And `reset` method is correctly called by `destroy` method.

The real problem seems to be that `destroy` is not called.

The real fix in this case would be to have `dealloc` calling `reset`.

Reply:
Or maybe we can can just call `destroy` in the `dealloc`?
Do you want to try this change?

What was the problem in the first place btw?

Reply:
In first place, the problem was that observers were not removed.

In my fork (Tchap app), that's what I implemented in the end : calling `destroy` in `dealloc`.

I will push a commit doing it.

Reply:
> In first place, the problem was that observers were not removed.

Sure. I meant: what problems caused these pending observers?

> I will push a commit doing it.

Thanks. It looks sensible to me.

Reply:
The pending observers didn't caused any visible problem.

That's just that i constated, working on this piece of code, adding myself new observers, that these observers weren't removed. The code wasn't called when tracing it.

Coding iOS apps for 15 years, I know it's not clean and can cause weird problems difficult to trace at some point.

Reply:
New push calling [self destroy] in dealloc.

Reply:
@alfogrillo could you review this PR please ?